start work on `delete_cassettes()`

maelle opened the pull request:
Fix #552

Not ready, will need a second pass

maelle committed:
start work on `delete_cassettes()`

sckott commented:
@maelle nice! should i look for the current state of this branch or do you want to do more first?

maelle committed:
no purrr

maelle committed:
anonymous

maelle committed:
new testthat

maelle commented on tests/testthat/_snaps/delete_cassettes.md:
```diff
@@ -0,0 +1,88 @@
+      * 'api-get.yml'
+      * 'api-post.yml'
+
+# delete_cassettes() works with exact cassette name
```
maybe we should remove this test, not very useful is it

maelle committed:
rm useless tests

maelle committed:
oops (from use_standalone)

maelle commented:
A bit confused by the test error that is unrelated to my changes?!

```r
 ══ Failed tests ════════════════════════════════════════════════════════════════
── Failure ('test-lightswitch.R:16:5'): turned_off works if and only if no cassettes active ──
Adding new snapshot:
Code
  turned_off(5 + 5)
Condition
  Error in `turn_off()`:
  ! You must eject all cassettes before you can turn vcr off.
Backtrace:
    ▆
 1. └─base::eval(...)
 2.   └─base::eval(...)
 3.     └─testthat::expect_snapshot(turned_off(5 + 5), error = TRUE) at test-lightswitch.R:16:5

[ FAIL 1 | WARN 0 | SKIP 1 | PASS 598 ]
Error:
! Test failures.
Execution halted
```

maelle commented:
Same failure in https://github.com/ropensci/vcr/pull/554 that has no real edit, maybe due to the new testthat version?

maelle commented:
Ah yes

> Additionally, [expect_snapshot()](https://testthat.r-lib.org/reference/expect_snapshot.html) and friends will now fail when creating a new snapshot on CI, as that’s usually a signal that you’ve forgotten to run the snapshot code locally before committing.

I'll make a commit to main!

maelle committed:
test: add missing snapshot file

sckott commented:
great i'll take a look soon!

sckott reviewed (changes requested):
Looks good so far. I'm still poking around. 

Shouldn't there be a `man/delete_cassettes.Rd` file? Can you add that please

maelle committed:
run document()

sckott committed:
Merge branch 'main' into delete

sckott committed:
potential fix for delete fxn when type = tests

- if cassette_path gives NULL abort with message
- add small test to cover this use case

sckott committed:
factor out test route to fxn in switch statement

sckott committed:
Merge pull request #557 from ropensci/delete-tests-no-dir-configured

potential fix for delete fxn when type = tests

sckott committed:
Merge branch 'main' into delete
